File: src/backend/CustomerService/src/main/java/com/example/demo/entities/Reservation.java
package com.example.demo.entities;

import java.time.LocalDate;
import java.time.LocalTime;

import com.fasterxml.jackson.annotation.JsonBackReference;
import com.fasterxml.jackson.annotation.JsonIgnoreProperties;

import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.JoinColumn;
import jakarta.persistence.ManyToOne;
import jakarta.persistence.Table;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;


@Entity
@Table(name = "reservations")
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
public class Reservation {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    int rt_id;

    @ManyToOne
    @JoinColumn(name = "u_id", nullable = false)
    User user;

    @ManyToOne
    @JoinColumn(name = "t_id", nullable = false)
    @JsonIgnoreProperties({"reservations"}) // to avoid infinite recursion
    private DineTable dineTable;


    @Column(name = "seat_count", nullable = false)
    int seatCount;

    @Column(nullable = false)
    private LocalDate date;

    @Column(nullable = false)
    private LocalTime time;
}

File: src/backend/dotNet/restaurant_management_system/waiter_srv/Models/Reservation.cs
using System;
using System.Collections.Generic;

namespace waiter_srv.Models;

public partial class Reservation
{
    public int RtId { get; set; }

    public int TId { get; set; }

    public int UId { get; set; }

    public TimeOnly Time { get; set; }

    public DateOnly Date { get; set; }

    public int SeatCount { get; set; }

    public virtual DineTable TIdNavigation { get; set; } = null!;

    public virtual User UIdNavigation { get; set; } = null!;
}
